multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/auditlogging/impl/DBAppender.java

```diff
@@ -1,43 +1,55 @@
 package org.cloudfoundry.multiapps.controller.core.auditlogging.impl;
 
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.filter.ThresholdFilter;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+import org.cloudfoundry.multiapps.controller.core.auditlogging.UserInfoProvider;
+import org.cloudfoundry.multiapps.controller.core.util.UserInfo;
+
 import java.sql.Connection;
 import java.sql.PreparedStatement;
 import java.sql.SQLException;
 import java.util.Objects;
-
 import javax.sql.DataSource;
 
-import org.apache.log4j.spi.LoggingEvent;
-import org.cloudfoundry.multiapps.controller.core.auditlogging.UserInfoProvider;
-import org.cloudfoundry.multiapps.controller.core.util.UserInfo;
-
-class DBAppender extends org.apache.log4j.AppenderSkeleton implements org.apache.log4j.Appender {
+class DBAppender extends AbstractAppender {
 
     interface LogEventAdapter {
-        void eventToStatement(String category, LoggingEvent event, UserInfo userInfo, PreparedStatement statement) throws SQLException;
+        void eventToStatement(String category, LogEvent event, UserInfo userInfo, PreparedStatement statement) throws SQLException;
     }
 
     interface ExceptionHandler {
         void handleException(Exception e);
     }
 
+    private static final Level LEVEL = Level.INFO;
+    private static final ThresholdFilter FILTER = ThresholdFilter.createFilter(LEVEL, Filter.Result.ACCEPT, Filter.Result.ACCEPT);
+    private static final PatternLayout LAYOUT = PatternLayout.createDefaultLayout();
+    private static final String DEFAULT_NAME = "DBAppender";
+
     private final DataSource dataSource;
     private final LogEventAdapter eventAdapter;
     private final String sql;
     private final ExceptionHandler exceptionHandler;
     private final UserInfoProvider userInfoProvider;
+    private final String appenderName;
 
     DBAppender(DataSource dataSource, String sql, LogEventAdapter eventAdapter, ExceptionHandler exceptionHandler,
-               UserInfoProvider userInfoProvider) {
+               UserInfoProvider userInfoProvider, String appenderName) {
+        super(DEFAULT_NAME, FILTER, LAYOUT, false, null);
         this.dataSource = Objects.requireNonNull(dataSource);
         this.sql = Objects.requireNonNull(sql);
         this.eventAdapter = Objects.requireNonNull(eventAdapter);
         this.exceptionHandler = Objects.requireNonNull(exceptionHandler);
         this.userInfoProvider = userInfoProvider;
+        this.appenderName = appenderName != null ? appenderName : DEFAULT_NAME;
```

Comment:
What names can we expect here?

Reply:
All the potential values for 'appenderName' are 3 (because they are hard coded).
I agree that this null-check is unnecessary and it is no longer present.